pkg/catalog/catalog.go

```diff
@@ -2868,6 +2869,78 @@ func (c *Catalog) checkCommitIDDuplication(ctx context.Context, repository *grav
 	return err
 }
 
+func (c *Catalog) GetPullRequest(ctx context.Context, repositoryID string, pullRequestID string) (*graveler.PullRequest, error) {
+	pid := graveler.PullRequestID(pullRequestID)
+	if err := validator.Validate([]validator.ValidateArg{
+		{Name: "name", Value: repositoryID, Fn: graveler.ValidateRepositoryID},
+		{Name: "pullRequestID", Value: pid, Fn: graveler.ValidatePullRequestID},
+	}); err != nil {
+		return nil, err
+	}
+
+	repository, err := c.getRepository(ctx, repositoryID)
+	if err != nil {
+		return nil, err
+	}
+
+	pr, err := c.Store.GetPullRequest(ctx, repository, pid)
+	if err != nil {
+		return nil, err
+	}
+
+	return pr, nil
+}
+
+func (c *Catalog) CreatePullRequest(ctx context.Context, repositoryID string, request *PullRequest) (string, error) {
+	srcBranchID := graveler.BranchID(request.SourceBranch)
+	destBranchID := graveler.BranchID(request.DestinationBranch)
+	if err := validator.Validate([]validator.ValidateArg{
+		{Name: "repository", Value: repositoryID, Fn: graveler.ValidateRepositoryID},
+		{Name: "dest", Value: destBranchID, Fn: graveler.ValidateBranchID},
+		{Name: "src", Value: srcBranchID, Fn: graveler.ValidateBranchID},
+	}); err != nil {
+		if errors.Is(err, graveler.ErrInvalidBranchID) {
```

Comment:
Why do you wrap `graveler.ErrInvalidBranchID`, but not the repo validation errors?

Reply:
Honestly the code was copied from CreateBranch. We don't need to wrap it - removing